include/cutlass/gemm/collective/intel_pvc_mma.hpp

```diff
@@ -184,72 +195,79 @@ struct CollectiveMma<
     (void)thread_idx;
     (void)smem_buf;
 
-    static_assert(is_rmem<FrgTensorD>::value, "D tensor must be rmem resident.");
-    static_assert(is_tuple<typename TensorA::engine_type::iterator::value_type>::value, "A tensor must be a tuple iterator.");
-    static_assert(is_tuple<typename TensorB::engine_type::iterator::value_type>::value, "B tensor must be a tuple iterator.");
-    static_assert(is_rmem<FrgTensorC>::value, "C tensor must be rmem resident.");
-
-    // Tensor to hold input data
-    constexpr int version =
-        is_same_v<GmemTiledCopyB, XE_2D_U16x16x16x2x1_V> ? 1 : 2;
-
-    Tensor tAr = make_tensor<typename TiledMma::ValTypeA>(Shape<Int<get<0>(SubgroupTileShape{}) * FragsK>, Int<1>>{});
-    Tensor tBr = make_tensor<typename TiledMma::ValTypeB>(Shape<Int<get<2>(SubgroupTileShape{}) * version>, Int<FragsN / version>>{});
-
-    Tensor tAr_view = make_tensor(static_cast<decltype(tAr) &&>(tAr).data(),
-                            Shape<Int<VecA>, Int<FragsM>, Int<FragsK>>{});
-    Tensor tBr_view = make_tensor(static_cast<decltype(tBr) &&>(tBr).data(),
-                            Shape<Int<VecB>, Int<FragsN>, Int<FragsK>>{},
-                            Stride<_1, Int<VecB * FragsK>, Int<VecB>>{});
-
-    // Instantiate the M MA object
+    // Instantiate the MMA object
     TiledMma tiled_mma;
+    auto thread_mma = tiled_mma.get_slice(thread_idx);
+    Tensor tCrA_partition = thread_mma.partition_fragment_A(gA(_, _, 0));
+    Tensor tCrA = make_tensor(static_cast<decltype(tCrA_partition) &&>(tCrA_partition).data(),
+                              tCrA_partition.shape());
+    Tensor tCrB_partition = thread_mma.partition_fragment_B(gB(_, _, 0));
+    Tensor tCrB = make_tensor(static_cast<decltype(tCrB_partition) &&>(tCrB_partition).data(),
+                              make_shape(size<0>(tCrB_partition.shape()),
+                                         size<2>(tCrB_partition.shape()),
+                                         size<1>(tCrB_partition.shape())));
```

Comment:
why not using tCrB_partition directly? This reshape shouldn't be needed.

Reply:
Due to the hardware spec, we can not smoothly partition or retile directly from a tensor to another tensor view. More detailly, mma->copy, the direction of fragments will change

Reply:
The hardware spec is encoded on software on the trait definition. If the information on the trait is correct `thread_mma.partition_fragment_A` and `thread_mma.partition_fragment_B` will return the tensors with the right shape/stride so we don't need to reshape it here. 

Note that for `thread_mma.partition_fragment_A` the tensor shape is already correct. You should be able to do just 

```diff
-  Tensor tCrA_partition = thread_mma.partition_fragment_A(gA(_, _, 0));
-  Tensor tCrA = make_tensor(static_cast<decltype(tCrA_partition) &&>(tCrA_partition).data(),
-                              tCrA_partition.shape());
+  Tensor tCrA = thread_mma.partition_fragment_A(gA(_, _, 0));
```

Reply:
I think it is not trivial, because `partition_fragment_X` just allocates the registers, which will be sent to the `thread_copy.retiled_D`, xmx and block2d seems not matching well. We ever tried to directly use these APIs with some perfect traits design, but failed

Reply:
Yes, I think there is something wrong in the trait definition for the mma or/and copy operations. We can look into that in a follow PR.

Reply:
The root cause is the difference between Nvidia convention and Intel's.  [Coord](https://github.com/codeplaysoftware/cutlass-fork/blob/sycl-develop/include/cute/atom/copy_traits_xe.hpp#L83C9-L83C71)